src/main/java/org/assertj/core/api/junit/jupiter/SoftAssertionsExtension.java

```diff
@@ -0,0 +1,119 @@
+ * managed by this extension begins when a parameter of type {@code SoftAssertions}
+ * or {@code BDDSoftAssertions} is resolved for a {@code @BeforeEach} method,
+ * test method, or {@code @AfterEach} method. The scope of the instance ends after
+ * all {@code @AfterEach} methods have been executed for the corresponding test.
+ * When the scope ends, {@code assertAll()} will be invoked on the instance to
+ * verify that no soft assertions failed.
+ *
+ * <h3>Example with {@code SoftAssertions}</h3>
+ *
+ * <pre class="code">
+ * {@literal @}ExtendWith(SoftAssertionsExtension.class)
+ * class ExampleTestCase {
+ * 
+ *    {@literal @}Test
+ *    void multipleFailures(SoftAssertions softly) {
+ *       softly.assertThat(2 * 3).isEqualTo(0);
+ *       softly.assertThat(Arrays.asList(1, 2)).containsOnly(1);
+ *       softly.assertThat(1 + 1).isEqualTo(2);
+ *    }
+ * }</pre>
+ *
+ * <h3>Example with {@code BDDSoftAssertions}</h3>
+ *
+ * <pre class="code">
+ * {@literal @}ExtendWith(SoftAssertionsExtension.class)
+ * class ExampleTestCase {
+ * 
+ *    {@literal @}Test
+ *    void multipleFailures(BDDSoftAssertions softly) {
+ *       softly.then(2 * 3).isEqualTo(0);
+ *       softly.then(Arrays.asList(1, 2)).containsOnly(1);
+ *       softly.then(1 + 1).isEqualTo(2);
+ *    }
+ * }</pre>
+ *
+ * @author Sam Brannen
+ * @since 3.13
+ */
+public class SoftAssertionsExtension implements ParameterResolver, AfterEachCallback {
+
+	@Override
+	public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext) {
+		boolean isTestLevel = extensionContext.getTestMethod().isPresent();
+		boolean isSoftAssertionsParameter = parameterContext.getParameter().getType() == SoftAssertions.class;
+		boolean isBddSoftAssertionsParameter = parameterContext.getParameter().getType() == BDDSoftAssertions.class;
+		return (isTestLevel && (isSoftAssertionsParameter || isBddSoftAssertionsParameter));
+	}
+
+	@Override
+	public Object resolveParameter(ParameterContext parameterContext, ExtensionContext extensionContext) {
+		if (parameterContext.getParameter().getType() == SoftAssertions.class) {
+			return getStore(extensionContext).getOrComputeIfAbsent(SoftAssertions.class);
+		}
+		return getStore(extensionContext).getOrComputeIfAbsent(BDDSoftAssertions.class);
+	}
+
+	@Override
+	public void afterEach(ExtensionContext extensionContext) throws Exception {
+		SoftAssertions softAssertions = getStore(extensionContext).remove(SoftAssertions.class, SoftAssertions.class);
+		if (softAssertions != null) {
+			softAssertions.assertAll();
+		}
+
+		BDDSoftAssertions bddSoftAssertions = getStore(extensionContext).remove(BDDSoftAssertions.class,
+			BDDSoftAssertions.class);
+		if (bddSoftAssertions != null) {
+			bddSoftAssertions.assertAll();
+		}
+	}
+
+	private Store getStore(ExtensionContext extensionContext) {
+		return extensionContext.getStore(Namespace.create(getClass(), extensionContext.getUniqueId()));
```

Comment:
Would `extensionContext.getStore(Namespace.create(getClass()));` be enough ? 

`getStore` being an `ExtensionContext` method it has access to the `getUniqueId()`, why do we need to pass it ?

Reply:
I already changed that in my latest commit. 😇